js/views/MapPage.js

```diff
@@ -56,11 +56,18 @@ const MapPage = React.createClass({
       this.active = true;
       this.generateTriggerMarkers(response);
 
-      setTimeout(() => {
-        if (!this.cancelCallbacks) {
-          this.setState({stage: 'ready'});
-        }
-      }, 300);
+      getUserLocationData((location) => {
+        setTimeout(() => {
```

Comment:
`InteractionManager` replacement I explained to you in a previous PR.
https://facebook.github.io/react-native/docs/interactionmanager.html